src/main/java/seedu/address/storage/JsonAdaptedNextOfKin.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.NextOfKin;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Relationship;
+
+/**
+ * Jackson-friendly version of {@link NextOfKin}.
+ */
+class JsonAdaptedNextOfKin {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+
+    private final String nextOfKinName;
+    private final String nextOfKinPhone;
+    private final String nextOfKinRelationship;
+
+    /**
+     * Constructs a {@code JsonAdaptedNextOfKin} with the given {@code nextOfKinName},
+     * {@code nextOfKinPhone} and {@code nextOfKinRelationship}.
+     */
+    @JsonCreator
+    public JsonAdaptedNextOfKin(@JsonProperty("nextOfKinName") String nextOfKinName,
+                                @JsonProperty("nextOfKinPhone") String nextOfKinPhone,
+                                @JsonProperty("nextOfKinRelationship") String nextOfKinRelationship) {
+        this.nextOfKinName = nextOfKinName;
+        this.nextOfKinPhone = nextOfKinPhone;
+        this.nextOfKinRelationship = nextOfKinRelationship;
+    }
+
+    /**
+     * Converts a given {@code NextOfKin} into this class for Jackson use.
+     */
+    public JsonAdaptedNextOfKin(NextOfKin source) {
+        if (source.isPresent()) {
+            this.nextOfKinName = source.getName().fullName;
+            this.nextOfKinPhone = source.getPhone().value;
+            this.nextOfKinRelationship = source.getRelationship().relationship;
+        } else {
+            this.nextOfKinName = null;
+            this.nextOfKinPhone = null;
+            this.nextOfKinRelationship = null;
```

Comment:
Good work! follows the format of other JsonAdaptable objects

Reply:
Thanks! :)